src/aleph_client/commands/domain.py

```diff
@@ -0,0 +1,313 @@
+    table.add_column("DNS TYPE", justify="right", style="cyan", no_wrap=True)
+    table.add_column("DNS NAME", style="magenta")
+    table.add_column("DNS VALUE", justify="right", style="green")
+
+    owner = account.get_address()
+    dns_rules = domain_validator.get_required_dns_rules(fqdn, target, owner)
+    for rule_id, rule in enumerate(dns_rules):
+        table.add_row(str(rule_id), rule.dns.get("type"), rule.dns.get("name"), rule.dns.get("value"))
+
+    console.print(table)
+
+    msg_status = "[bold green]Detecting dns..."
+
+    with console.status(msg_status) as status:
+        max_retries = 5
+        while dns_rules:
+            rule = dns_rules[0]
+            """Get rules check status"""
+            checks = await check_domain_records(fqdn, target, owner)
+            completed_rules = []
+            for index, rule in enumerate(dns_rules):
+                if checks[rule.name] is True:
+                    """Pass configured rules"""
+                    completed_rules.append(rule)
+                    console.log(f"record: {index} [bold green] OK")
+
+            for _rule in completed_rules:
+                dns_rules.remove(_rule)
+                completed_rules = []
+
+            if dns_rules:
+                rule = dns_rules[0]
+                console.log(f"[green]{rule.info}")
+                status.update(f"{msg_status} [bold red]{rule.on_error}")
+
+                max_retries -= 1
+                sleep(10)
+
+            if max_retries == 0:
+                status.stop()
+                continue_ = Confirm.ask("Continue?")
+                if continue_:
+                    status.start()
+                    max_retries = 5
+                else:
+                    raise typer.Exit()
+
+    """Attach option"""
+    if Confirm.ask(f"Attach ressource to [bold green]{fqdn}"):
+        await attach_resource(account, fqdn, item_hash)
+
+    raise typer.Exit()
+
+
+@app.command()
+@coro
+async def attach(
+    private_key: Optional[str] = typer.Option(
+        sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY
+    ),
+    private_key_file: Optional[Path] = typer.Option(
+        sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE
+    ),
+    fqdn: str = typer.Argument(..., help=help_strings.CUSTOM_DOMAIN_NAME),
+    item_hash: Optional[str] = typer.Option(None, help=help_strings.CUSTOM_DOMAIN_ITEM_HASH),
+):
+    """Attach resource to a Custom Domain."""
+    account: AccountFromPrivateKey = _load_account(private_key, private_key_file)
+
+    await attach_resource(account, fqdn, item_hash)
+    raise typer.Exit()
+
+@app.command()
+@coro
+async def detach(
+    private_key: Optional[str] = typer.Option(
+        sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY
+    ),
+    private_key_file: Optional[Path] = typer.Option(
+        sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE
+    ),
```

Comment:
Just noticing: We start to get quite a bit of duplication here. Is it possible to define options at a higher level and have them apply to all the commands?